### ISSUE TITLE
Update Carbon

### ISSUE THREAD

**Issue by AlanGreene** [tektoncd/dashboard#3493]
Update to Carbon 11.

Carbon 10 is being sunset and will no longer be supported as of 1st October 2024.

Announcements:
- https://medium.com/carbondesign/announcing-the-carbon-design-system-release-schedule-f33144e9141
- https://medium.com/carbondesign/moving-forward-on-deprecating-carbon-components-and-carbon-components-react-4f2f0c3d8448

We have already done quite a bit of preparation for this, including simplification of the data / API layers which allowed us to reduce the number of custom page containers used in the app in https://github.com/tektoncd/dashboard/issues/2452 and related issues.

So far it looks like React 18 should not be required for this effort 🤞 If it is, see https://github.com/tektoncd/dashboard/issues/3492 for details.

Migration guide: https://carbondesignsystem.com/migrating/guide/overview/

Initially we'll migrate to the new versions keeping the design as close as possible to existing. This is to minimise the impact / overhead on tests, developer experience, and user experience during the migration process.

Some changes will be required, particularly around use of notifications and tooltips. Details in the migration guide.

Once the initial migration is complete we can consider adopting some of the new components and behaviours provided to enhance the experience.

**Comment by AlanGreene** [comment]
Results of first pass at updating the Dashboard to Carbon 11 (most styling is expected to be broken / missing for now):
- run the provided code mods to automatically migrate as much as possible
- disable any remaining problematic code / styles
  - comment out use of legacy tokens in the styles
  - comment out use of legacy components
  - minimal changes to get the app to build and run

This was just to get a baseline status and a better feel for the amount of work required to migrate fully.

~10% of unit tests (55/581) are failing in this state, affecting ~25% of the test suites (19/73).

About page:
![image](https://github.com/tektoncd/dashboard/assets/2829095/209cfcf6-3754-4885-849e-74b425c8f3d0)

PipelineRuns:
![image](https://github.com/tektoncd/dashboard/assets/2829095/93032b22-0f71-4884-8343-5728b2660746)

PipelineRun details:
![image](https://github.com/tektoncd/dashboard/assets/2829095/95001739-86b4-4e9e-8cd5-cf1f1e24fb13)

The next step is to go through the longer / more detailed manual migration guide at https://github.com/carbon-design-system/carbon/blob/main/docs/migration/v11.md to resolve the styling and functional issues.

There are a number of breaking changes to components such a notifications and tooltips that will need to be addressed, as well as changes to the structure of tabs, etc.

My hope is that the bulk of the issues will be resolved by changes to common components that are reused in multiple places.

**Comment by AlanGreene** [comment]
Looking a bit better after some minor changes to the theme-related styles and how the base Carbon component styles were being loaded.

About:
![image](https://github.com/tektoncd/dashboard/assets/2829095/5ad1295b-036d-48a8-9539-3449276c130d)

PipelineRuns:
![image](https://github.com/tektoncd/dashboard/assets/2829095/74e9bd99-e349-4946-8d48-3557bc893992)

PipelineRun details:
![image](https://github.com/tektoncd/dashboard/assets/2829095/b3f2f453-34e0-4a1b-b01b-5e818cec6dd0)

There are still a number of layer / theme related issues that need to be resolved, but looks like we have a solid foundation to start from.

One issue that may take a little more experimentation (and was also present after the first pass above) is an apparent change in behaviour for the side nav. It now overlaps the content when open instead of influencing (pushing) the grid:
![image](https://github.com/tektoncd/dashboard/assets/2829095/1e290430-d523-43ac-95e9-094a3e31d257)

**Comment by AlanGreene** [comment]
Getting closer, still some colour, typography, and spacing tokens to migrate but this should be the bulk of the big styling issues resolved.

About:
![image](https://github.com/tektoncd/dashboard/assets/2829095/70796384-6a5e-4dfe-909d-1db18bb83603)

PipelineRuns:
![image](https://github.com/tektoncd/dashboard/assets/2829095/7c191058-4dfe-4922-853b-5cd8f291c353)

Next, move onto the functional issues and get the tests passing again.

**Comment by AlanGreene** [comment]
Starting to look a lot better now. Most major styling issues resolved, working through remaining cleanup and functional fixes.

PipelineRuns:
![image](https://github.com/tektoncd/dashboard/assets/2829095/3424e560-fa37-4ae6-9429-feb74c686209)

PipelineRun details:
![image](https://github.com/tektoncd/dashboard/assets/2829095/cbd8d29c-4a99-41a2-b51c-9f959e2b9c5f)

There are still some minor issues with the actions menu button and the log toolbar buttons but everything on this page appears to be functional at least 😸

**Comment by AlanGreene** [comment]
With the addition of https://github.com/tektoncd/dashboard/pull/3501/commits/01b9c370c28f11fc6a83cddf4a1896df3b295138 the E2E tests are now passing again 🎉 

There are still some styling and accessibility issues to be resolved, as well as some remaining functional issues not currently covered by E2E tests.

**Comment by AlanGreene** [comment]
a11y issues with Accordion component:
- https://github.com/carbon-design-system/carbon/issues/16781
- https://github.com/carbon-design-system/carbon/issues/16413

There's also an issue with the ActionableNotification component's close button trapping focus.
I've worked around it for now using the 'unstable' + experimental feature flags 😅 : https://github.com/tektoncd/dashboard/pull/3501/commits/74f2dc96deb5eacc6424403713f01572ea65e6cc#diff-9b40aa71e20dfce56078f07084c09937debb4b201f87f9d965de48b520418c23R21-R23

ComboBox behaviour has changed in v11 too, no longer firing the onChange handler when the 'clear' button is clicked. Worked around it on the global NamespacesDropdown by listening for the underlying DownShift state change events but would be nice to have a cleaner solution similar to the one we had with Carbon v10. Slack thread open with the Carbon folks discussing options.

Also noticed the ComboBox clear behaviour is not quite as expected on the 'create' pages, even though the event is being handled and the URL is updated as expected, the input is not cleared consistently. Blurring, re-focusing / opening, and closing the listbox again seems to trigger it to update correctly, so something's interfering with the first update. Warnings in the console with details of the underlying DownShift component changing from uncontrolled to controlled, may not be related, but should also be resolved.

There's some odd behaviour with the actions menu on the run details pages too. Keyboard nav of the items in the open menu is causing the page to scroll. Not seeing the same behaviour in the Carbon storybook or our own for that matter, so likely something else on the page contributing to the problem.

Some good news is that most of the unit tests are now passing again, current status:
```
Test Files  9 failed | 64 passed (73)
Tests  16 failed | 565 passed | 1 skipped (582)
```

and we've reduced the size of the build output by disabling unused fonts.
